vips/image.go

```diff
@@ -1180,6 +1180,40 @@ func (r *ImageRef) ModulateHSV(brightness, saturation float64, hue int) error {
 	return nil
 }
 
+// Pixelate applies a simple pixelate filter to the image
+func (r *ImageRef) Pixelate(factor float64) error {
```

Comment:
do you want to assert that `factor > 0`?

Reply:
I feel like this is a little specific to hang on `ImageRef` type. 

How would you feel about making this a utility method that takes in (and maybe return a new ImageRef)? Or it can modify in place

Reply:
The reason I placed under ImageRef is that to my understanding the same effect can be archived by with a single pass on the image, and I didn't want to end up with two APIs that do the same.

Reply:
Sorry, all I'm saying is that we could easily do:

```
func Pixelate(image *ImageRef, factor float64) error {
  // your code
}
```

I mean, I know there's a bunch of things on ImageRef already, I do wish it were more minimal and the "modifiers" of it were separate functions. WDYT? Let me know and we can merge

Reply:
ok, this way we won't end up with BC later on.

Reply:
BC?

Reply:
Backwards Compatible :)

Reply:
> Sorry, all I'm saying is that we could easily do:
> 
> ```
> func Pixelate(image *ImageRef, factor float64) error {
>   // your code
> }
> ```
> 
> I mean, I know there's a bunch of things on ImageRef already, I do wish it were more minimal and the "modifiers" of it were separate functions. WDYT? Let me know and we can merge

Implemented